server/odc-service/src/main/java/com/oceanbase/odc/service/connection/table/TableService.java

```diff
@@ -126,11 +142,29 @@ public List<Table> list(@NonNull @Valid QueryTableParams params) throws SQLExcep
                         tableExtension);
             }
             if (types.contains(DBObjectType.EXTERNAL_TABLE)) {
-                generateListAndSyncDBTablesByTableType(params, database, dataSource, tables, conn,
-                        DBObjectType.EXTERNAL_TABLE, tableExtension);
+                if (checkExternalTableSupported(dataSource.getDialectType(), conn)) {
+                    generateListAndSyncDBTablesByTableType(params, database, dataSource, tables, conn,
+                            DBObjectType.EXTERNAL_TABLE, tableExtension);
+                }
             }
-            return tables;
         }
+        return tables;
+    }
+
+    @SkipAuthorize("permission check inside")
+    public boolean checkExternalTableSupported(@NonNull DialectType dialectType, @NonNull Connection conn)
```

Comment:
checkExternalTableSupported -> isExternalTableSupported

check prefix usually lead to an exception and 'is' prefix usually lead to a boolean

Reply:
ok